.vitepress/theme/components/Feedback/Feedback.vue

```diff
@@ -0,0 +1,317 @@
+<script setup lang="ts">
+import { watch, ref, nextTick } from 'vue';
+import { useRoute } from 'vitepress';
+import LoadingIcon from '@theme/components/Icons/LoadingIcon.vue';
+import { sendEvent } from '@theme/scripts/gtag';
+import CardSingle from '@theme/components/CardLists/CardSingle.vue';
+
+enum FeedbackState {
+  START = 'START',
+  YES = 'YES',
+  NO = 'NO',
+  END = 'END',
+}
+
+const route = useRoute();
+const currentState = ref(FeedbackState.START);
+const feedback = ref('');
+const feedbackTextarea = ref<HTMLInputElement | null>(null);
+const isSending = ref(false);
+const hasError = ref(false);
+
+function onButtonClick(value: FeedbackState.YES | FeedbackState.NO) {
+  // don't change state if opposite button is active.
+  const opposingState = value === FeedbackState.YES ? FeedbackState.NO : FeedbackState.YES;
+  if ([opposingState].includes(currentState.value)) {
+    return;
+  }
+  currentState.value = value;
+  nextTick(() => feedbackTextarea.value?.focus());
+}
+
+function cancelForm() {
+  currentState.value = FeedbackState.START;
+}
+
+function encode(data: Record<string, string>) {
+  return Object.keys(data)
+    .map((key) => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
+    .join('&');
+}
+
+// We send a form to Netlify: https://docs.netlify.com/forms/setup/
+function submitForm() {
+  isSending.value = true;
+  sendEvent({
+    eventName: 'sb_doc_feedback_helpful',
```

Comment:
Used to be `doc_feedback_helpful`, but since we have multiple docs, this now follows the pattern of
`wc_doc_feedback_helpful`
`sb_doc_feedback_helpful`
to be explicit.